openwisp-qa-check

```diff
@@ -44,29 +44,48 @@ show_help() {
   printf "  --csslinter\t\t\t: Runs prettier check on css files\n"
   printf "Jslinter check options:\n"
   printf "  --jslinter\t\t\t: Runs prettier check on javascript files\n"
+  printf "Ignore pattern options:\n"
+  printf "  --ignore-pattern\t\t: Specify file patterns to ignore\n"
 }
 
 echoerr() { echo "$@" 1>&2; }
 
 runcheckendline() {
   flag=0
-  for i in $(find . -type f ! -path "*/*.egg-info/*" \
-    ! -path "./.*" \
-    ! -path "*.min.*" \
-    ! -path "*.svg" -exec grep -Iq . {} \; -and -print); do
-    if [ "$(tail -c 1 $i)" != "" ]; then
-      echo "$i needs newline at the end"
+  
+  # Start with basic exclusions
+  exclude_patterns=("./.*" "*/.git/*" "*/*.egg-info/*" "*.min.*" "*.svg")
+  
+  # Add user-provided ignore pattern if specified
+  if [ -n "$IGNORE_PATTERN" ]; then
+    echo "Ignoring: $IGNORE_PATTERN"
+    exclude_patterns+=("$IGNORE_PATTERN")
+  fi
+  
+  # Build the exclusion arguments
+  exclude_args=""
+  for pattern in "${exclude_patterns[@]}"; do
+    exclude_args="$exclude_args -path '$pattern'"
```

Comment:
```suggestion
    exclude_args="$exclude_args -not -path \"$pattern\""
```
any specific reason for using single quotes, This will result in literal single quotes in the command string.